test/test_sparse.py

```diff
@@ -1121,37 +1121,47 @@ def do_test(t):
         do_test(self.SparseTensor())
 
     def _test_resize_shape(self, x_i, x_v, x_size, y_i, y_v, y_size):
-        x = torch.sparse_coo_tensor(torch.zeros(x_i), torch.randn(x_v), torch.Size(x_size))
+        x_v_numel = torch.zeros(x_v).numel()
+        y_v_numel = torch.zeros(y_v).numel()
+        x = torch.sparse_coo_tensor(torch.zeros(x_i),
+                                    torch.arange(x_v_numel).resize_(x_v).to(torch.float),
+                                    torch.Size(x_size))
         x_dense = x.to_dense()
-        y = torch.sparse_coo_tensor(torch.zeros(y_i), torch.randn(y_v), torch.Size(y_size))
+        y = torch.sparse_coo_tensor(torch.zeros(y_i),
+                                    torch.ones(y_v).to(torch.float),
+                                    torch.Size(y_size))
         y_dense = y.to_dense()
         x.resize_as_(y)
         x_dense.resize_as_(y_dense)
         self.assertEqual(x.shape, y.shape)
         self.assertEqual(x._sparseDims(), y._sparseDims())
         self.assertEqual(x._denseDims(), y._denseDims())
-        self.assertEqual(x.to_dense(), x_dense)
+        # Here we make sure that the original data are preserved after resizing
+        self.assertEqual(x.to_dense().view(-1)[0:x_v_numel].view(x_v),
+                         x_dense.view(-1)[0:x_v_numel].view(x_v))
```

Comment:
nice!  how about also checking that the shape of x and y match up with the shape of x_dense and y_dense?